app/src/main/res/values/strings.xml

```diff
@@ -236,14 +236,14 @@
     <string name="unsupport_magisk_msg">This version of the app does not support Magisk version lower than %1$s.\n\nThe app will behave as if no Magisk is installed, please upgrade Magisk as soon as possible.</string>
     <string name="unsupport_general_title">Abnormal State</string>
     <string name="unsupport_system_app_msg">Running this app as a system app is not supported. Please revert the app to a user app.</string>
-    <string name="unsupport_other_su_msg">An \"su\" command that does not belong to Magisk is detected. Please remove the other unsupported su.</string>
+    <string name="unsupport_other_su_msg">Please uninstall the \"su\" rooting-solution competing with Magisk from your device. Reinstalling Magisk will also do this.</string>
     <string name="unsupport_external_storage_msg">Magisk is installed to external storage. Please move the app to internal storage.</string>
-    <string name="unsupport_nonroot_stub_msg">The app cannot continue to work in the hidden state as root was lost. Please restore it back to the original APK.</string>
+    <string name="unsupport_nonroot_stub_msg">The hidden Magisk app stopped working because root was lost. Please restore the original Magisk APK.</string>
```

Comment:
It is clear what the hidden app is
What is being restored is clear
It is clear what the original APK is of (could have been the rooting app)
Exactly as long as the suggested alternative.
`The hidden app cannot continue to work because root was lost. Please restore it to the original APK.

Reply:
```suggestion
    <string name="unsupport_nonroot_stub_msg">The hidden Magisk app stopped working because root was lost. Please revert it to the original Magisk APK.</string>
```

Reply:
This is more wordy again, restore works better. Stop changing things just to change them.